src/recipe/recipe.controller.spec.ts

```diff
@@ -1,18 +1,175 @@
 import { Test, TestingModule } from '@nestjs/testing';
+import { RecipeService } from './recipe.service';
 import { RecipeController } from './recipe.controller';
+import { PrismaService } from 'src/prisma/prisma.service';
 
-describe('RecipeController', () => {
-  let controller: RecipeController;
+describe("Recipe Controller's tests", () => {
+    let recipeController: RecipeController;
+    let recipeService: RecipeService;
+    let prismaService: PrismaService;
 
-  beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      controllers: [RecipeController],
-    }).compile();
+    const mockRecipe = {
+        id: expect.any(Number),
+        createdAt: expect.any(Number),
+        title: expect.any(String),
+        description: expect.any(String),
+        ingredients: expect.any(String),
+        preparation: expect.any(String),
+    };
 
-    controller = module.get<RecipeController>(RecipeController);
-  });
+    class mockRecipeRequest {
+        title: string;
+        description: string;
+        ingredients: string;
+        preparation: string;
+    }
 
-  it('should be defined', () => {
-    expect(controller).toBeDefined();
-  });
+    const mockRecipeService = {
+        createRecipe: jest.fn((dto: mockRecipeRequest) => {
+            return {
+                id: 1,
+                createdAt: Date.now(),
+                ...dto,
+            };
+        }),
```

Comment:
you can shorten this method (but don't have to!) like this:
```suggestion
        createRecipe: jest.fn((dto: mockRecipeRequest) => ({
                id: 1,
                createdAt: Date.now(),
                ...dto,
            })
        ),
```